Description: Provide --no-ssl2 option to disable ssl2 methods
Bug-Debian: http://bugs.debian.org/589706
Bug-Debian: http://bugs.debian.org/622065
Bug-Node: https://github.com/joyent/node/issues/880
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2011-04-14
--- a/src/node_crypto.cc
+++ b/src/node_crypto.cc
@@ -75,11 +75,23 @@
 
     String::Utf8Value sslmethod(args[0]->ToString());
     if (strcmp(*sslmethod, "SSLv2_method") == 0)
+#ifndef OPENSSL_NO_SSL2
       method = SSLv2_method();
+#else
+      return ThrowException(Exception::Error(String::New("SSLv2 methods disabled")));
+#endif
     if (strcmp(*sslmethod, "SSLv2_server_method") == 0)
+#ifndef OPENSSL_NO_SSL2
       method = SSLv2_server_method();
+#else
+      return ThrowException(Exception::Error(String::New("SSLv2 methods disabled")));
+#endif
     if (strcmp(*sslmethod, "SSLv2_client_method") == 0)
+#ifndef OPENSSL_NO_SSL2
       method = SSLv2_client_method();
+#else
+      return ThrowException(Exception::Error(String::New("SSLv2 methods disabled")));
+#endif
     if (strcmp(*sslmethod, "SSLv3_method") == 0)
       method = SSLv3_method();
     if (strcmp(*sslmethod, "SSLv3_server_method") == 0)
--- a/wscript
+++ b/wscript
@@ -58,6 +58,12 @@
                 , dest='without_ssl'
                 )
 
+  opt.add_option( '--no-ssl2'
+                , action='store_true'
+                , default=False
+                , help="Disable OpenSSL v2"
+                , dest='openssl_nov2'
+                )
 
   opt.add_option('--shared-v8'
                 , action='store_true'
@@ -193,6 +199,9 @@
         conf.fatal("Could not autodetect OpenSSL support. " +
                    "Make sure OpenSSL development packages are installed. " +
                    "Use configure --without-ssl to disable this message.")
+    # Disable ssl v2 methods
+    if o.openssl_nov2:
+      conf.env.append_value("CPPFLAGS", "-DOPENSSL_NO_SSL2=1")
   else:
     Options.options.use_openssl = conf.env["USE_OPENSSL"] = False
 
