Description: v8 3.8 compatibility
 MayContainNonAscii is not available in libv8 3.8.
Author: Jérémy Lal <kapouer@melix.org>
Forwarded: not-needed
Last-Update: 2012-09-30
--- a/src/stream_wrap.cc
+++ b/src/stream_wrap.cc
@@ -310,11 +310,12 @@
       break;
 
     case kUtf8:
-      if (!(string->MayContainNonAscii())) {
+      /* if (!(string->MayContainNonAscii())) { // this call is not libv8 3.8-compatible
         // If the string has only ascii characters, we know exactly how big
         // the storage should be.
         storage_size = string->Length();
-      } else if (string->Length() < 65536) {
+      } else */
+      if (string->Length() < 65536) {
         // A single UCS2 codepoint never takes up more than 3 utf8 bytes.
         // Unless the string is really long we just allocate so much space that
         // we're certain the string fits in there entirely.
