From: =?UTF-8?q?J=C3=A9r=C3=A9my=20Lal?= <kapouer@melix.org>
Date: Fri, 1 Jan 2010 22:22:12 +0100
Subject: [PATCH] rename node.1 to nodejs.1

---
 wscript |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

--- a/wscript
+++ b/wscript
@@ -559,8 +559,8 @@
 
   # Only install the man page if it exists.
   # Do 'make doc install' to build and install it.
-  if os.path.exists('doc/node.1'):
-    bld.install_files('${PREFIX}/share/man/man1/', 'doc/node.1')
+  if os.path.exists('doc/nodejs.1'):
+    bld.install_files('${PREFIX}/share/man/man1/', 'doc/nodejs.1')
 
   bld.install_files('${PREFIX}/bin/', 'bin/*', chmod=0755)
   bld.install_files('${PREFIX}/share/nodejs/wafadmin', 'tools/wafadmin/*.py')
